### PR TITLE
fix(broker): flush write if any entry was appended

### LINKED ISSUE #6784: Follower doesn't flush written entries if append is only partially written
**Describe the bug**
The follower only flushes the appended entries if every entry in the append request is written. 

**Expected behavior**
If any entry in the append request is written, then the journal the write should be flushed. 

**Log/Stacktrace**
None.

**Environment:**
 - OS: [e.g. Linux]
 - Zeebe Version: [e.g. 0.20.0]
 - Configuration: [e.g. exporters etc.]

### PR DESCRIPTION
**Pull request by miguelpires**
## Description

The follower only flushed the appended entries if every entry in the AppendRequest was written. This commit makes the follower flush if any entry was written.

## Related issues

closes #6784 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [x] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java [hunk 1]
```diff
@@ -531,6 +531,7 @@ protected void appendEntries(
 
         final boolean failedToAppend = tryToAppend(future, reader, entry, index, lastEntry);
         if (failedToAppend) {
+          flush(lastLogIndex - 1, request.prevLogIndex());
```
Comment (deepthidevaki) [inline atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java:534]: The future is already completed before the log is flushed. That means, the follower acknowledges before the entries are flushed. `failAppend()` must be called after this flush. Can you check if it possible to refactor `tryToAppend` and take `failAppend` out of it and move it here?

**Review by deepthidevaki — CHANGES_REQUESTED** [review]
We should ensure that all entries are flushed before sending the ack to the leader.

File: atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java (hunk 1 above)
Comment (MiguelPires) [reply atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java:534]: It's completed but not returned so the ack happens after the flush

Comment (deepthidevaki) [reply atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java:534]: Ya. You are right. The usage for future there is confusing. It is a synchronous execution even though it uses a future :slightly_smiling_face:

**Review by deepthidevaki — APPROVED** [review]
:100: 
bors r+

File: atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java (hunk 1 above)
Comment (MiguelPires) [reply atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java:534]: I agree, it's a bit strange to use a future but return it synchronously. It's a good candidate for a future refactor

**Pull request merged**